proteus/mprans/BoundaryConditions.py

```diff
@@ -584,10 +641,20 @@ def ux_dirichlet(x, t):
 
             self.u_dirichlet.uOfXT = get_inlet_ux_dirichlet(0)
             self.v_dirichlet.uOfXT = get_inlet_ux_dirichlet(1)
-            self.w_dirichlet.uOfXT = get_inlet_ux_dirichlet(2) 
+            self.w_dirichlet.uOfXT = get_inlet_ux_dirichlet(2)
+            self.u_diffusive.resetBC()
+        
+        if kInflow is not None:
+            self.k_dirichlet.uOfXT = inlet_k_dirichlet
+            self.k_advective.resetBC()
+            self.k_diffusive.resetBC()
+        if dissipationInflow is not None:
+            self.dissipation_dirichlet.uOfXT = inlet_dissipation_dirichlet
+            self.dissipation_advective.resetBC()
+            self.dissipation_diffusive.resetBC()     
 
 
-# FOLLOWING BOUNDARY CONDITION IS UNTESTED #
+# FOLLOWING BOUNDARY CONDITIO IS UNTESTED #
```

Comment:
Is CONDITIO Italian:)

Reply:
It's latin actually ahah :)

Reply:
Just one thing Chris. This morning I had a meeting with @adimako . 
We agreed on the best configuration to use at this stage. I'll pull soon it.
I'd like if you can still give a look deeper in the extraction module for velocity (and kappa as well). Randomly I get nan value when I use a big number of processors.